### ISSUE TITLE
Feature request: Ability to have a SuppressionDescriptor disabled by default

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#51278]
Suggests the following changes to the public api surface

```diff
namespace Microsoft.CodeAnalysis
{
    public sealed class SuppressionDescriptor : IEquatable<SuppressionDescriptor?>
    {
        public string Id { get; }
        public string SuppressedDiagnosticId { get; }
        public LocalizableString Justification { get; }
+        public bool IsEnabledByDefault { get; }

        public SuppressionDescriptor(
            string id,
            string suppressedDiagnosticId,
-            string justification);
+            string justification,
+            bool isEnabledByDefault = true);

        public SuppressionDescriptor(
            string id,
            string suppressedDiagnosticId,
-            LocalizableString justification);
+            LocalizableString justification,
+            bool isEnabledByDefault = true);
    }
}
```

**Comment by jinujoseph** [comment]
cc @mavasani

**Comment by sharwell** [comment]
This would also need the ability to enable it, which I'm not sure is possible today.

**Comment by roji** [comment]
Is this needed, given that the suppressor can look into .editorconfig for options as standard?

**Comment by Youssef1313** [comment]
> Is this needed, given that the suppressor can look into .editorconfig for options as standard?

- The implementation doesn't seem to be trivial in this case.
- Also this won't be "by default" behavior.

**Comment by roji** [comment]
> The implementation doesn't seem to be trivial in this case.

Is it any less trivial than what the [Roslyn analyzers](https://github.com/dotnet/roslyn-analyzers) already do? I agree it could be nice for analyzers to have some infrastructure around accessing editorconfig options, but that seems better than to add a new concept of "enabled by default".

> Also this won't be "by default" behavior.

Why not? The suppressor can decide to not suppress unless it sees some option, no?

**Comment by roji** [comment]
Note that specifically for #21663, I don't think this is necessary anymore - we've decided to suppress by default, and one can opt-out with a ruleset.

**Comment by Youssef1313** [comment]
Yeah I see it's not needed for https://github.com/dotnet/efcore/issues/21663. But as a general feature, I think it's useful to have.
